numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -3297,15 +3297,14 @@ PyArray_Where(PyObject *condition, PyObject *x, PyObject *y)
         NPY_ITER_READONLY | NPY_ITER_ALIGNED
     };
     common_dt = PyArray_ResultType(2, &op_in[2], 0, NULL);
+    if (common_dt == NULL) {
+        goto fail;
+    }
     PyArray_Descr * op_dt[4] = {common_dt, PyArray_DescrFromType(NPY_BOOL),
                                 common_dt, common_dt};
     NpyIter * iter;
     NPY_BEGIN_THREADS_DEF;
 
-    if (common_dt == NULL || op_dt[1] == NULL) {
```

Comment:
I think you still need to check `op_dt[1] != NULL` right after the definition of `op_dt` - though really that should not fail...

Reply:
OK, makes sense! If it fails, there are certainly worse problems afoot...